### ISSUE TITLE
Java Agent FY25Q4 - Support Adding Labels on APM Logs

### ISSUE THREAD

**Issue by kford-newrelic** [newrelic/newrelic-java-agent#2037]
## Is your feature request related to a problem? Please describe.
Customer switched from using Logstash to forward logs to New Relic, to using our Java agent to do the same thing.  

## Feature Description
In their Logstash config, they were able to tag on custom attributes to their log events, using the following format:
```
input {
  file {
    path => "/var/log/supervisor/java-*.log"
    add_field => {
      custom_field => "custom_value"
    }
start_position => "beginning"

codec => json {
    }
  }
}
```
They would like to see us implement a similar feature in the Java agent config, not unlike our current Labels solution but specific to Log events.  A simple list of key / value pairs that would be attached to every log event sent up by that Java agent.

## Describe Alternatives
N/A

## Additional context
Before starting any initial spiking, we should try to engage the requesting customer (through their account team), to see if they'd be willing to guide feature development and later try snapshot agent versions.

Review epic description on [NR-315883](https://new-relic.atlassian.net/browse/NR-315883)

Similar Jiras on this topic - [NR-303269](https://new-relic.atlassian.net/browse/NR-303269), [NR-310852](https://new-relic.atlassian.net/browse/NR-310852) & [NR-312135](https://new-relic.atlassian.net/browse/NR-312135)

## Priority
Please help us better understand this feature request by choosing a priority from the following options: 
[Nice to Have] 

[NR-303269]: https://new-relic.atlassian.net/browse/NR-303269?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

[NR-315883]: https://new-relic.atlassian.net/browse/NR-315883?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

**Comment by jasonjkeller** [comment]
The Java agent supports adding custom attributes to log events via MDC. https://docs.newrelic.com/docs/apm/agents/java-agent/configuration/java-agent-configuration-config-file/#log-context-data
